features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Listing/components/ThumbnailView.jsx

```diff
@@ -0,0 +1,323 @@
+        color: theme.palette.common.white,
+    },
+});
+
+/**
+ * Slide up transition for modal
+ *
+ * @param {any} props Properties
+ * @returns {Slide} Slide up transition
+ */
+function Transition(props) {
+    return <Slide direction='up' {...props} />;
+}
+
+/**
+ * Provides a view for the API Thumbnail image.
+ * This can be either user defined image uploaded earlier or a generated Image.
+ */
+class ThumbnailView extends Component {
+    /**
+     * Initializes the ThumbnailView Component
+     *
+     * @param {any} props Component properties
+     */
+    constructor(props) {
+        super(props);
+        this.state = { open: false, file: null, thumbnail: null };
+        this.handleClick = this.handleClick.bind(this);
+        this.handleClose = this.handleClose.bind(this);
+    }
+
+    /**
+     * Load required data for showing the thubnail view
+     */
+    componentDidMount() {
+        const thumbApi = new Api();
+        const { api } = this.props;
+        thumbApi.getAPIThumbnail(api.id)
+            .then((response) => {
+                if (response && response.data && response.data.size > 0) {
+                    const url = windowURL.createObjectURL(response.data);
+                    this.setState({ thumbnail: url });
+                }
+            });
+    }
+
+    /**
+     * Event listener for file drop on the dropzone
+     *
+     * @param {File} acceptedFile dropped file
+     */
+    onDrop(acceptedFile) {
+        this.setState({ file: acceptedFile });
+    }
+
+    /**
+     * @param {SyntheticEvent} e React event object
+     */
+    handleClick(e) {
+        if (e.target.id === 'btnEditAPIThumb') {
+            this.setState({ open: true });
+        } else if (e.target.id === 'btnUploadAPIThumb') {
+            const { api } = this.props;
+            this.uploadThumbnail(api.id, this.state.file[0]);
+        }
+    }
+
+    /**
+     * Add new thumbnail image to an API
+     *
+     * @param {String} apiId ID of the API to be updated
+     * @param {File} file new thumbnail image file
+     */
+    uploadThumbnail(apiId, file) {
+        const { intl } = this.props;
+        if (!apiId || !file) {
+            Alert.error(intl.formatMessage({ id: 'thumbnail.validation.error' }));
+            return;
+        }
+        const api = new Api();
+        const thumbnailPromise = api.addAPIThumbnail(apiId, file);
+        thumbnailPromise.then(() => {
+            Alert.info(intl.formatMessage({ id: 'thumbnail.upload.success' }));
+            this.setState({ open: false, thumbnail: file.preview });
+        }).catch((error) => {
+            if (process.env.NODE_ENV !== 'production') {
+                console.log(error);
+            }
+            Alert.error(intl.formatMessage({ id: 'thumbnail.upload.error' }));
+        });
+    }
+
+    /**
+     * Handle modal close event
+     */
+    handleClose() {
+        if (this.state.file) {
+            windowURL.revokeObjectURL(this.state.file.preview);
```

Comment:
@praminda  Shall we call this `revokeObjectURL` method in `componentwillunmount` lifecycle method 

> Perform any necessary cleanup in this method, such as invalidating timers, canceling network requests, or cleaning up any subscriptions that were created in 

~([DOC](https://reactjs.org/docs/react-component.html#componentwillunmount))

Reply:
shall we do this only for actual thumbnail image, not the dropped file preview? Because if we wait until component unmount to revoke dropped file preview, the link will be there even after closing the dialog. WDYT?

Reply:
shall we do this only for actual thumbnail image, not the dropped file preview? Because if we wait until component unmount to revoke dropped file preview, the link will be there even after closing the dialog. WDYT?